File: Python/1704. Determine if String Halves Are Alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        half = len(s) // 2
        i, j, val = 0, half, 0

        for _ in range(half):
            if s[i] in 'aeiouAEIOU':
                val += 1

            if s[j] in 'aeiouAEIOU':
                val -= 1

            i += 1
            j += 1

        return val == 0

File: C++/1704. Determine if String Halves Are Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s)
    {
        int half = s.size() / 2, i = 0, j = half, val = 0;


        for (;i < half;++i, ++j)
        {
            if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u' || s[i] == 'A' || s[i] == 'E' || s[i] == 'I' || s[i] == 'O' || s[i] == 'U')
                ++val;

            if (s[j] == 'a' || s[j] == 'e' || s[j] == 'i' || s[j] == 'o' || s[j] == 'u' || s[j] == 'A' || s[j] == 'E' || s[j] == 'I' || s[j] == 'O' || s[j] == 'U')
                --val;
        }

        if (val != 0)
            return false;
        return true;

    }
};